eslint/tslint behavior difference with comma-dangle

### Bug Report (delete this for feature requests)

- __TSLint version__: `3.7.3`
- __TypeScript version__: `1.8.9`
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
function f() {
  return [1, {
    a: 2,
    b: 2,
  }];
}
```

with `tslint.json`:

```json
{
  "rules": {
    "trailing-comma": [
      true,
      {
        "multiline": "always",
        "singleline": "never"
      }
    ]
  }
}
```

#### Actual behavior

```
repro.ts[5, 3]: missing trailing comma
```

#### Expected behavior

(no error)

When I run this code through eslint with the following config:

```javascript
{
  "rules": {
    'comma-dangle': [2, 'always-multiline'],
  }
}
```

it passes. But tslint wants me to write this instead:

```ts
function f() {
  return [1, {
    a: 2,
    b: 2,
  },];  // <---
}
```

which I don't think makes much sense. In fact, eslint complains about it:

```
repro.ts
  5:4  error  Unexpected trailing comma  comma-dangle
```

--- a/src/rules/trailingCommaRule.ts
+++ b/src/rules/trailingCommaRule.ts
@@ -53,13 +53,16 @@ class TrailingCommaWalker extends Lint.RuleWalker {
     private lintNode(node: ts.Node) {
         const child = node.getChildAt(1);
         if (child != null && child.kind === ts.SyntaxKind.SyntaxList) {
-            const isMultiline = node.getText().match(/\n|\r/);
-            const option = this.getOption(isMultiline ? "multiline" : "singleline");
             const grandChildren = child.getChildren();
 
             if (grandChildren.length > 0) {
                 const lastGrandChild = grandChildren[grandChildren.length - 1];
                 const hasTrailingComma = lastGrandChild.kind === ts.SyntaxKind.CommaToken;
+
+                const endLineOfNode = this.getSourceFile().getLineAndCharacterOfPosition(node.getEnd()).line;
+                const endLineOfLastElement = this.getSourceFile().getLineAndCharacterOfPosition(lastGrandChild.getEnd()).line;
+                const isMultiline = endLineOfNode !== endLineOfLastElement;
+                const option = this.getOption(isMultiline ? "multiline" : "singleline");
 
                 if (hasTrailingComma && option === "never") {
                     this.addFailure(this.createFailure(lastGrandChild.getStart(), 1, Rule.FAILURE_STRING_NEVER));

--- a/test/rules/trailing-comma/multiline-always/test.ts.lint
+++ b/test/rules/trailing-comma/multiline-always/test.ts.lint
@@ -12,7 +12,6 @@ var x = [{
     c: (a + b)
              ~ [missing trailing comma]
 }];
-~   [missing trailing comma]
 
 var s = {
     sA: 6,

--- a/test/rules/trailing-comma/multiline-never/test.ts.lint
+++ b/test/rules/trailing-comma/multiline-never/test.ts.lint
@@ -5,7 +5,6 @@ var v = [{
     c: (a + b),
               ~ [trailing comma]
 },];
- ~   [trailing comma]
 
 var x = [{
     a: 1,

--- a/test/rules/trailing-comma/singleline-always/test.ts.lint
+++ b/test/rules/trailing-comma/singleline-always/test.ts.lint
@@ -11,6 +11,7 @@ var x = [{
     d: 34,
     c: (a + b)
 }];
+~   [missing trailing comma]
 
 var s = {
     sA: 6,

--- a/test/rules/trailing-comma/singleline-never/test.ts.lint
+++ b/test/rules/trailing-comma/singleline-never/test.ts.lint
@@ -4,6 +4,7 @@ var v = [{
     d: 34,
     c: (a + b),
 },];
+ ~   [trailing comma]
 
 var x = [{
     a: 1,

